File: Beginner/Python/Array/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
            :type nums: List[int]
            :type target: int
            :rtype: List[int]
            """
        L = []
        for index1, number1 in enumerate(nums):
            for index2, number2 in enumerate(nums):
                if number1 + number2 == target and index1 != index2:
                    L.append(index1)
                    L.append(index2)
                    return L
        return L

File: Beginner/Java/Array/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        //         method 1
        // int[] res = new int[2];
        // for (int i = 0; i < nums.length; i ++)
        //     for (int j = i + 1; j < nums.length; j ++) {
        //         if (nums[i] + nums[j] == target) {
        //             res[0] = i; res[1] = j;
        //             return res;
        //         }
        //     }
        // return res;
        
        //         method 2
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i ++) {
            if (map.containsKey(target - nums[i])) {
                return new int[] {map.get(target - nums[i]), i};
            }
            map.put(nums[i], i);
        }
        return new int[] {0, 0};
    }
}
